### PR TITLE
ACLs in Swarm see #1199

### PR DESCRIPTION
**Pull request by doronp**
Authorization (ACLs) support in Swarm as a first step towards multi tenancy. #1199 

This PR includes (organized by Commits for convenience)

1 - Validation API & Default implementation- Approve/Disapprove/condition of 'Who wants to to what'?
2 - Handler API & Default implementation - Use labels to enforce decision of 1
3 - Feature flag --multiTenant to get this feature going (regular Swarm otherwise) 
4 - Sign off :-)

### CODE REVIEW

**Comment by chanwit** [comment]
Thank your @doronp for this PR. Could you please sign the commits as suggested by Gordon?

File: test/authZ/simpleTest.sh [hunk 1]
```diff
@@ -0,0 +1,91 @@
+SwarmNode="http://127.0.0.1:2379"
```
Comment (chanwit) [inline test/authZ/simpleTest.sh]: We do testing with `bats`. Could you please convert this test to `bats`?

File: cli/commands.go [hunk 2]
```diff
@@ -28,7 +28,8 @@ var (
 				flTLS, flTLSCaCert, flTLSCert, flTLSKey, flTLSVerify,
 				flHeartBeat,
 				flEnableCors,
-				flCluster, flDiscoveryOpt, flClusterOpt},
+				flCluster, flDiscoveryOpt, flClusterOpt,
+				multiTenant},
```
Comment (chanwit) [inline cli/commands.go]: We have a convention to declare this with the `fl` prefix.

File: api/primary.go [hunk 3]
```diff
@@ -6,6 +6,7 @@ import (
 
 	log "github.com/Sirupsen/logrus"
 	"github.com/docker/swarm/cluster"
+	"github.com/docker/swarm/pkg/authZ"
```
Comment (chanwit) [inline api/primary.go]: It's better to `godep` this package rather than putting it directly under `pkg/authZ`.

**Comment by abronan** [comment]
Thanks @doronp! This is huge :)

This will not make it for 1.0 but we'll take time to review and get this merged for 1.1 if we all good.

In the meantime, can you please sign-off all your commits so that it passes the DCO check? Sorry about that!

File: api/primary.go (hunk 3 above)
Comment (abronan) [inline api/primary.go]: I guess it's fine to have it under `pkg` as this one is created for the purpose of the PR. Having it under Godep would mean having it under a separate repo, but we want to be able to fix things asap without having to update dependencies with Godeps at first (can be considered later on but for now it's best to keep it under `pkg` for now imho).

**Comment by doronp** [comment]
Thanks for your comments. I've Opened PR again (for internal PM reasons) see #1366 
I've started working on the comments I've got - Some of them already in #1366 
Thanks
Doron P

**Pull request closed without merging**